sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -730,6 +734,175 @@ public virtual Response<RecognizeEntitiesResultCollection> RecognizeEntitiesBatc
+        public virtual async Task<RecognizeCustomEntitiesOperation> RecognizeCustomEntitiesAsync(
+            WaitUntil waitUntil,
+            IEnumerable<TextDocumentInput> documents,
+            string projectName,
+            string deploymentName,
+            RecognizeCustomEntitiesOptions options = default,
+            CancellationToken cancellationToken = default)
+        {
+            return await _serviceClient.RecognizeCustomEntitiesAsync(waitUntil, documents, projectName, deploymentName, options, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Runs a predictive model to identify a collection of custom named entities
+        /// in the passed-in documents, and categorize those entities into custom types
+        /// such as contracts or financial documents.
+        /// <para>For more information on available categories, see
+        /// <see href="https://docs.microsoft.com/azure/cognitive-services/language-service/custom-named-entity-recognition/overview"/>.</para>
+        /// <para>For a list of languages supported by this operation, see
+        /// <see href="https://aka.ms/talangs"/>.</para>
+        /// <para>For document length limits, maximum batch size, and supported text encoding, see
+        /// <see href="https://aka.ms/azsdk/textanalytics/data-limits"/>.</para>
+        /// </summary>
+        /// <param name="documents">The documents to analyze.</param>
+        /// <param name="projectName">The name of the project which owns the models being consumed.</param>
+        /// <param name="deploymentName">The name of the deployment being consumed.</param>
+        /// <param name="language">The language that all the documents are
+        /// written in. If unspecified, this value will be set to the default
+        /// language in <see cref="TextAnalyticsClientOptions.DefaultLanguage"/> in the request
+        /// sent to the service. If set to an empty string, the service will apply a model
+        /// where the language is explicitly set to "None".</param>
+        /// <param name="options"><see cref="TextAnalyticsRequestOptions"/> used to
+        /// select the version of the predictive model to run, and whether
+        /// statistics are returned in the response.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/>
+        /// controlling the request lifetime.</param>
+        /// <returns>A result containing the collection of entities identified
+        /// for each of the documents, as well as scores indicating the confidence
+        /// that a given entity correctly matches the identified substring.</returns>
+        /// <exception cref="RequestFailedException">Service returned a non-success
+        /// status code.</exception>
+        /// <exception cref="NotSupportedException">This method is only supported in service API version 2022-05-01 and newer.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
For example, this method could call a base `RecognizeCustomEntities` passing a `WaitUntil.Started` so you don't have to dup implementation. It would have to pass the scope. Might be worth seeing which is less overhead to maintain.

Reply:
See my reply to the other comment.